Print the last acked position when a semi-sync slave connects to the master

percona-ysorokin opened the pull request:
Summary:
This information will be helpful in debugging.

Reference Patch: facebook/mysql-5.6@dc0940c

Originally Reviewed By: guokeno0

hermanlee commented on sql/binlog.cc:
```diff
@@ -8789,7 +8789,11 @@ void MYSQL_BIN_LOG::process_after_commit_stage_queue(THD *thd, THD *first) {
+      const char *file_name;
+      head->get_trans_pos(&file_name, &pos, nullptr, nullptr);
+      // NO_LINT_DEBUG
+      sql_print_information("Signalling async threads to pos: %s:%llu",
```
Doesn't this trigger printing to the error log after each group commit? We're only looking to print out the async threads position when a semi-sync slave connects to the server.

hermanlee replied:
Indeed, what I'm seeing in the error log for each insert statement on the master:

2020-01-10T02:21:00.973077Z 8 [Note] [MY-010462] [Repl] Start binlog_dump to master_thread_id(8) slave_server(2), pos(, 4)
2020-01-10T02:21:11.128362Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:374
2020-01-10T02:21:13.446666Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:648
2020-01-10T02:21:14.938622Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:922
2020-01-10T02:21:34.745018Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:1196
2020-01-10T02:21:36.329375Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:1470
2020-01-10T02:21:45.940968Z 9 [Note] [MY-000000] [Server] Signalling async threads to pos: master-bin.000001:1744

hermanlee reviewed (changes requested):
This needs to be fixed so that the logging only happens when a semi-sync slave connects to the server.

percona-ysorokin commented:
@hermanlee Got you. Will rework the fix.

percona-ysorokin commented:
PR updated

avbelov23 committed:
Print the last acked position when a semi-sync slave connects to the master

Summary:
This information will be helpful in debugging.

Reference Patch: facebook/mysql-5.6@dc0940c

Originally Reviewed By: guokeno0

percona-ysorokin commented:
PR updated one more time (even better implementation)

percona-ysorokin commented:
PR updated one more time (Better function names)

hermanlee commented on sql/mysqld.cc:
```diff
@@ -6108,6 +6108,7 @@ static int init_server_components() {
 #endif
     locked_in_memory = 0;
 
+  mysql_bin_log.print_last_acked_pos();
```
It seems like this function is only called during server initialization, vs. whenever a semi-sync slave connects to the server?

It seems like ReplSemiSyncMaster::add_slave() would be a better place for this call?

hermanlee commented:
It would be fine to drop this PR.
